extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisReceiver.scala

```diff
@@ -119,7 +132,9 @@ private[kinesis] class KinesisReceiver(
    */
   override def onStart() {
     workerId = InetAddress.getLocalHost.getHostAddress() + ":" + UUID.randomUUID()
-    credentialsProvider = new DefaultAWSCredentialsProviderChain()
+    if(credentialsProvider == null) {
+      credentialsProvider = new DefaultAWSCredentialsProviderChain()
```

Comment:
Can this simply be the default value of the field rather than delay its init? or is this expensive?